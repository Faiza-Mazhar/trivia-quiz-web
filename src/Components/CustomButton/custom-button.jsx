import React from "react";
import "./custom-button.style.scss";

const CustomButton = ({
  children,
  isGoogleSignIn,
  isAnswerButton,
  ...otherProps
}) => {
  const classNames = () => {
    let classNames = "custom-button";

    if (isGoogleSignIn) {
      classNames += " google-sign-in";
    }

    if (isAnswerButton) {
      classNames += " answer";
    }

    return classNames;
  };
  return (
    <button className={classNames()} {...otherProps}>
      {children}
    </button>
  );
};
export default CustomButton;
