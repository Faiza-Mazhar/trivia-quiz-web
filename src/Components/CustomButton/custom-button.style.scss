.custom-button {
  width: 100%;
  color: white;
  letter-spacing: 0.5px;
  line-height: 50px;
  background-color: #556342;
  height: 50px;
  font-weight: bolder;
  border: none;
  font-family: "Open Sans Condensed", sans-serif;
  cursor: pointer;
  border-radius: 5px;

  &:hover {
    background-color: #2a3121;
    color: white;
    border: 1px solid black;
  }

  &.google-sign-in {
    background-color: #4285f4;
    color: white;

    &:hover {
      background-color: #357ae8;
      color: black;
      border: none;
    }
  }

  &.answer {
    background-color: #e2eee2;
    color: #2a3121;

    &:focus {
      background-color: #2a3121;
      color: white;
    }
  }
}
