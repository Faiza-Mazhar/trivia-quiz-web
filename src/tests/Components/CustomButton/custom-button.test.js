import CustomButton from "../../../Components/CustomButton/custom-button";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";

afterEach(cleanup);

test("custom button is rendered with the test defined as children", () => {
  const buttonText = "Click Me";
  const { getByText } = render(<CustomButton>{buttonText}</CustomButton>);

  expect(getByText(/Click Me/i).textContent).toBe(buttonText);
});

test("Button rendered", () => {
  const component = renderer.create(<CustomButton>Click Me</CustomButton>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Button class changes the class it is isGoogleSignedIn", () => {
  const component = renderer.create(
    <CustomButton isGoogleSignedIn>Click Me</CustomButton>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("CustomButton rendered the google icon", () => {
  const component = renderer.create(
    <CustomButton isGoogleSignedIn>
      <img alt="google" />
      Click Me
    </CustomButton>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
