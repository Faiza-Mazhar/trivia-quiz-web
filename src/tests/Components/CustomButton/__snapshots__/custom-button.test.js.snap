// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button class changes the class it is isGoogleSignedIn 1`] = `
<button
  className=" custom-button"
  isGoogleSignedIn={true}
>
  Click Me
</button>
`;

exports[`Button rendered 1`] = `
<button
  className=" custom-button"
>
  Click Me
</button>
`;
